include/revng/Yield/Calls/Slices.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Yield/Graph.h"
+
+class MetaAddress;
+
+namespace yield::calls {
+
+Graph calleePreservingSlice(const Graph &Input,
+                            const MetaAddress &SlicePoint = MetaAddress());
+Graph callerPreservingSlice(const Graph &Input,
+                            const MetaAddress &SlicePoint = MetaAddress());
```

Comment:
What are these function doing? Can you please document them?

Reply:
What do you think? https://github.com/revng/revng/pull/191/commits/381d2caf195d296fa3fb439d4d4fda0a2476b239

Reply:
I'm not totally sold on the "forward facing slice" terminology, but overall the example in the comments help.

But the example in the comment really seems to be only about the `calleePreservingSlice`. So now I ask: what's the difference between the `calleePreserving` and `callerPreserving` version? Can you explain that in the comment too? I'd really like that if someone reads just the comments already has a good understanding of what these functions do, without having to look at the implementation. 

I think we need to find a better naming for the functions too, that explains what they do. `Slice` suggests that they do something on the graph, but as you noted in the comment they don't.  If I understand correctly what these functions do, they generate a new graph from the `Input` such that the result is a Tree, not a Graph. Is that right? Can you propose some names based on this?

Reply:
> So now I ask: what's the difference between the `calleePreserving` and `callerPreserving` version?

Yeah, sorry. The only difference is that one is an `llvm::Inverse` of the other. I'll change it to reflect that more clearly.

> `Slice` suggests that they do something on the graph

Well, they do slice out a piece of the graph to export. But I agree there should be a better name for them out there.

> Can you propose some names based on this?

The only things that come to mind is along the lines of `subtree` or `subgraph` or something like that. Do you have any suggestions/ideas?

Reply:
> Well, they do slice out a piece of the graph to export.

Do they? It seems to me that they take the original graph, they change it, and return something else. The graph they return cannot be obtained from the original graph by just removing edges or nodes, which is what I would expect from something called "slice". IIUC you also duplicate nodes and move edges to the duplicated nodes here, so you're not effectively just "slicing" the graphs. Am I wrong?

Reply:
Yeah, you're right! I got lost in the old naming scheme, since these didn't use to add anything new to the graph. We should definitely rename them now.

Reply:
> Yeah, sorry. The only difference is that one is an llvm::Inverse of the other. I'll change it to reflect that more clearly.

https://github.com/revng/revng/pull/191/commits/3dd085f68ade81361503b960b7cc11019e95f561

Reply:
Ok for the comments you added.

About naming, what about `makeSimpleCallerTree` and `makeSimpleCalleeTree`?

Reply:
Sounds good to me! @aleclearmind, what do you think?

Reply:
If I understand what these are correctly, what do you think of `makeCallTree` and `makeInverseCallTree`?
What @pfez suggested works too.

Reply:
I prefer using `Caller` and `Callee` as opposed to pushing `Inverse` thing in. I'm not attached to the `Simple` term though, so `makeCallerTree` and `makeCalleeTree` would also work just fine.

Reply:
Done: https://github.com/revng/revng/pull/191/commits/b373b7bb9c88f0c55f0d6ec49ff61dc96f5f01bd